tests/main/lxd-mount-units/task.yaml

```diff
@@ -0,0 +1,38 @@
+summary: Test mount units generated for snaps inside lxd container are correct.
+
+details: |
+    Test that mount units generated by snapd-generator inside lxd container
+    are correct for snapfuse.
+
+# only 20.10+, we want lxd images that come with snaps preinstalled.
+systems: [ubuntu-20.10*]
+
+execute: |
+    echo "Install lxd"
+    snap install lxd
+
+    lxd waitready
+    lxd init --auto
+
+    VERSION_ID="$(. /etc/os-release && echo "$VERSION_ID" )"
+    lxd.lxc launch --quiet "ubuntu:$VERSION_ID" ubuntu
+
+    lxd.lxc file push --quiet "$GOHOME"/snapd_*.deb "ubuntu/root/"
```

Comment:
we have some logic inside another script for setting up a container with the new snapd, is there a reason we're not using that here? I'm worried that for some reason something here will eventually need internet access inside the container and if we're not setting up the proxy it will fail, etc. or there will be issues installing the snapd deb here that are avoided with the workarounds in that script etc.

To be clear, this isn't a blocker, I'm just worried that this test might be a bit flaky without those extra things setup.

Reply:
The reason I didn't use that script (in fact I tried to add my checks first to existing lxd test, but then decided to add a new test) is it's a bit blunt and first purges snapd (and with that removes all snaps, e.g. core18 that come preseeded with lxd). I can add http_proxy logic here but I wasn't sure what is the expected use case for proxy?

Reply:
thanks for explaining that, makes sense now.

as to the proxy, I'm not sure, I always seem to see random lxd things fail around internet issues so I just implicitly assume that _things_ in the container need internet access and as such setting up a proxy makes sense, but perhaps this case is straight forward enough it won't need a proxy setup at all.